### PR TITLE
prov/efa: Fail unit test early if resource struct fails

### PR DESCRIPTION
**Pull request by sunkuamzn**
No point in continuing the test or checking the return value in the test body

Signed-off-by: Sai Sunku <sunkusa@amazon.com>

### CODE REVIEW

File: prov/efa/test/efa_unit_test_common.c [hunk 1]
```diff
@@ -87,11 +87,13 @@ int efa_unit_test_resource_construct(struct efa_resource *resource, enum fi_ep_t
 	if (ret)
 		goto err;
 
-	assert_int_equal(ret, 0);
```
Comment (sunkuamzn) [inline prov/efa/test/efa_unit_test_common.c:90]: No need to assert return value here because of the `goto` statements above

**Comment by sunkuamzn** [comment]
In this PR, I only modified the cases where the return value from `efa_unit_test_resource_construct` was being checked. There are other cases where the return value was simply ignored. https://github.com/ofiwg/libfabric/search?q=efa_unit_test_resource_construct

It's in those cases that the early failure is helpful

File: prov/efa/test/efa_unit_test_ep.c [hunk 2]
```diff
@@ -23,8 +23,7 @@ void test_rxr_ep_pkt_pool_flags(struct efa_resource **state)
 	int ret;
```
Comment (wenduwan) [inline prov/efa/test/efa_unit_test_ep.c:23]: Now this is used variable.

Comment (sunkuamzn) [reply prov/efa/test/efa_unit_test_ep.c]: fixed

**Review by wenduwan — APPROVED** [review]
Thank you

**Review by lrbison — APPROVED** [review]

**Review by wzamazon — APPROVED** [review]

**Pull request merged**